### DISCUSSION TITLE
Support for New/Dynamic Telemetry Points

### DISCUSSION THREAD

**Discussion by sdu949877** [nasa/openmct#7046] [Community Q&A]
Hello,

I'm trying to make the OpenMCT domain object system support missions where the incoming telemetry data structure is dynamic. 

To this end, I'm trying to modify OpenMCT to support adding, modifying, and deleting telemetry and folder domain objects built by the dictionary plugin. For example, when OpenMCT receives a telemetry point that does not have a corresponding domain object defined by the dictionary plugin, the domain object tree would automatically add a new domain object without needing a modal to prompt the user for inputs or refresh the web page to reload the plugins. Any ideas on where to start?

Thank you!

**Comment by jvigliotta** [comment]
Hi @sdu949877,

I'm not sure how the telemetry is coming in, in your scenario, to my knowledge, you would need a corresponding domainObject to receive telemetry data using Open MCT. If all of your Telemetry is dynamic, you could have a "catchall" domain object as a root object that receives all your telemetry, then creates your folder domain objects on the fly and adds any new endpoints to their compositions. To add and remove folders, you could just add and remove those items from the composition of their parent objects.

Let me know if this answers your question, if not can you provide me with a little more background on how your telemetry is structured?
